### DISCUSSION TITLE
Recover from lost connection to Server

### DISCUSSION THREAD

**Discussion by gyachts** [rossmann-engineering/EasyModbusTCP.NET#66] [Q&A]
I have a test application that is running on a server and ingesting data to a SQL server and is working very well but I'm struggling to recover when the ethernet/network connection is interrupted or when the PLC is powered down and then restarted.
What is the best practice to detect and automatically recover in this case without user interaction with the application?  

Thank you.

Greg

**Comment by hulinning2** [comment]
I think you just try to connect again until you got the connection your application can start to consume the data again

**Comment by EugeneAmmosov** [comment]
Write PLC "heartbeat" logic that will change address or Teg every minute. In response your app will detect change, if nothing changed in one minute - assume connection lost and require reconnect.
